[Feature] follow 기본 기능 구현 및 api response dto, application.properties 수정

HaenaCho01 opened the pull request:
## 팔로우 기본 기능 구현

1. 사용자 간 팔로우
2. 사용자 간 언팔로우
3. 팔로우 한 사용자 목록 조회

* 팔로잉 한 사용자의 게시글 조회는 추후 게시글 기능 구현 완료 후 추가할 예정


## api response dto 위치 이동 및 내용 수정

* comment 패키지 -> common 패키지 이동 및 내용 수정


## application.properties 수정

* JWT 시크릿 키가 노출되지 않도록 하고, 각자의 인텔리제이 환경변수에 맞게 설정할 수 있도록 코드 수정

HaenaCho01 committed:
application.properties 수정

JWT 시크릿 키가 노출되지 않도록 하고, 각자의 인텔리제이 환경변수에 맞게 설정할 수 있도록 코드 수정

HaenaCho01 committed:
api response dto 위치 이동

comment 패키지 -> common 패키지

HaenaCho01 committed:
api response dto 위치 이동

comment 패키지 -> common 패키지

HaenaCho01 committed:
팔로우 기본 기능 구현

1. 사용자 간 팔로우
2. 사용자 간 언팔로우
3. 팔로우 한 사용자 목록 조회

* 팔로잉 한 사용자의 게시글 조회는 추후 게시글 기능 구현 완료 후 추가할 예정

ohsutinn commented on src/main/java/com/sparta/followfollowmeproject/follow/controller/FollowController.java:
```diff
@@ -0,0 +1,41 @@
+	}
+
+	// 선택한 user 팔로우
+	@PostMapping("/{followingId}")
```
api 작성간 기능이 명확히 드러났으면 좋겠습니다.
/{followingId}/follow 와 같은 방식이 좋을 거 같습니다 !
아래 unfollow 도 동일한 의견입니다 !

HaenaCho01 replied:
[[fix] 팔로우 기능 - Repository 에서](https://github.com/sepang-pang/8lowFollowMe_Project/pull/7/commits/1c22d8ad71d94a7ab783632d938128ac410e22ae) @repository [어노테이션 삭제 및 api 수정](https://github.com/sepang-pang/8lowFollowMe_Project/pull/7/commits/1c22d8ad71d94a7ab783632d938128ac410e22ae)

수정 완료했습니다! 감사합니다!👍

ohsutinn commented on src/main/java/com/sparta/followfollowmeproject/follow/repository/FollowRepository.java:
```diff
@@ -0,0 +1,16 @@
+
+import java.util.List;
+
+@Repository
```
@Repository 어노테이션을 달아줌으로써 명시적으로 역할을 드러내는 건 좋지만
이미 JpaRepository 에서 상속을 받기 때문에 해당 어노테이션은 팀원들과의 통일성 및 코드 구현 간 중복성 해결을 위해 제거하는 편이 좋을 거 같다는 생각입니다 !

HaenaCho01 replied:
[[fix] 팔로우 기능 - Repository 에서](https://github.com/sepang-pang/8lowFollowMe_Project/pull/7/commits/1c22d8ad71d94a7ab783632d938128ac410e22ae) @repository [어노테이션 삭제 및 api 수정](https://github.com/sepang-pang/8lowFollowMe_Project/pull/7/commits/1c22d8ad71d94a7ab783632d938128ac410e22ae)

수정 완료했습니다! 감사합니다!👍

HaenaCho01 committed:
[fix] 팔로우 기능 - Repository 에서 @Repository 어노테이션 삭제 및 api 수정

1. JpaRepository 상속 받으므로 Repository 에서 @Repository 어노테이션 삭제
2. api 작성 간 기능이 명확히 드러나도록 api 수정

HaenaCho01 committed:
[fix] 팔로우 기능 - 예외 처리 추가

### 아래의 예외 처리를 추가하여 commit 합니다.
1. 동일한 팔로우 관계가 두 번 이상 등록되지 않도록 예외 처리
2. 본인을 팔로우하고자 할 경우 예외 처리

HaenaCho01 commented:
@HaenaCho01
[[fix] 팔로우 기능 - 예외 처리 추가](https://github.com/sepang-pang/8lowFollowMe_Project/pull/7/commits/0e2ef2b00d8c29efa56e8f6ea89792f7817a862c)

### 아래의 예외 처리를 추가하여 commit 합니다.
1. 동일한 팔로우 관계가 두 번 이상 등록되지 않도록 예외 처리
2. 본인을 팔로우하고자 할 경우 예외 처리
